pulp_rpm/app/tasks/copy.py

```diff
@@ -60,8 +80,47 @@ def find_children_of_content(content, repository_version):
             except MultipleObjectsReturned:
                 raise
 
-    # TODO: Find rpms referenced by PackageGroups,
-    # PackageGroups referenced by PackageCategories, etc.
+    # PackageCategories & PackageEnvironments resolution must go before PackageGroups
```

Comment:
Let's use ```.iterator()```, since we only iterate once over the queryset we don't need it to be cached.

Reply:
I think there's a few more ways to improve this.  You could try something like this:

```python
group_names = set()
for packagecategory in packagecategories.select_related('group_ids'):
    for category_package_group in packagecategory.group_ids:
        group_names.add(category_package_group['name'])
packagegroups.union(PackageGroup.objects.filter(pk__in=packagegroup_ids, name__in=group_names).values_list('pk', flat=True)
```

That way there are a lot less queries.

Reply:
Unfortunately `'group_ids'` is not supported field for `select_related`, neither `prefetch_related` works. So will stay with original approach using `iterator()`.

Reply:
Can we still add a TODO note that in the future, we should look into performance improvement?  I'm sure it can be done (although it's not worth spending time on right now)